src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -749,6 +765,10 @@ private protected override void SaveModel(ModelSaveContext ctx)
             ctx.Writer.Write(_predictedLabelColumnName);
             ctx.Writer.Write(_learningRate);
             ctx.Writer.Write(_classCount);
+
+            for (int j = 0; j < _classCount; j++)
+                ctx.SaveNonEmptyString(_keyValueAnnotations[j]);
```

Comment:
This should be done only if `_keyValueAnnotations` is not null (it is possible that the training label column does not contain key values), and you need to serialize a byte before it indicating whether there are key values or not.